Source/Core/Core/HW/DVD/DVDInterface.cpp

```diff
@@ -822,6 +822,11 @@ void ExecuteCommand(ReplyType reply_type)
   if (static_cast<DICommand>(s_DICMDBUF[0] >> 24) != DICommand::RequestError)
     SetDriveError(DriveError::None);
 
+  if (DVDThread::GetDiscType() == DiscIO::Platform::Triforce)
+  {
+    s_DICMDBUF[0] <<= 24;
```

Comment:
This seems like an odd place to do this, with the exact value being read (and shifted right by 24) immediately before _and_ afterwards. Depending on what you want to achieve with this, it should probably a lot earlier, a lot later, or maybe even not at all.

Reply:
Triforce games send the DI commands in a different endian, so this swaps the endian back

Reply:
1. Shouldn't this be done _before_ line 822 checks for `DICommand::RequestError`?
2. This doesn't swap endinanness, it just shifts the value and discards 3 bytes. You might wanna use `Common::swap32` instead then (especially for clarity on what is going on.)
3. It seems weird that it uses a different endianness compared to the GC, but if this is really the case, I'd like to see a comment highlighting this (mainly because _I_ didn't know about this, and I'm fairly certain other devs also don't)

Reply:
Here's a rough outline of what happens (with the session key blanked out), using F-Zero AX (Rev C) as an example:
```
Triforce game sends read command: 0x****00a8
Command is left shifted: a8000000
Command is right shifted: a8 (Command is now normal)
```
I will add a comment though, and yeah it probably should be before 822.

Reply:
Using Common::swap32 means that the randomised bytes arent zeroed out, leading to commands being incorrectly interpreted as read subcommands, which causes issues.

Reply:
Then it _really_ needs a comment when the bytes are _supposed to be_ zeroed out.